src/docfx/lib/log/Error.cs

```diff
@@ -47,12 +47,16 @@ public Error(ErrorLevel level, string code, string message, FilePath? file = nul
             Name = name;
         }
 
-        public Error WithCustomError(CustomError customError)
+        public Error WithCustomRule(CustomRule customRule, bool? isCanonicalVersion = null)
         {
+            var level = isCanonicalVersion != null && customRule.CanonicalVersionOnly
```

Comment:
nit: consider split it into 2 lines, one is to apply `customRule.Severity` override, the other to apply `isCanonical` override.